src/Workspaces/Core/Portable/Storage/SQLite/SQLitePersistentStorage_WriteBatching.cs

```diff
@@ -120,7 +120,8 @@ internal partial class SQLitePersistentStorage
             return;
 
             // Local functions
-            async Task<(Task previousTask, TaskCompletionSource<int> taskCompletionSource)> GetWriteTaskAsync()
+            //[PerformanceSensitive("https://github.com/dotnet/roslyn/issues/36114", OftenCompletesSynchronously = true)]
+            async ValueTask<(Task previousTask, TaskCompletionSource<int> taskCompletionSource)> GetWriteTaskAsync()
```

Comment:
any thought on making an instance-method so it doesn't accidentally capture?

Reply:
Will be done in a follow-up PR.